packages/adblocker-webextension/adblocker.ts

```diff
@@ -37,6 +37,41 @@ type StreamFilter = WebRequest.StreamFilter & {
   onerror: (event: any) => void;
 };
 
+function isFirefox() {
+  try {
+    return navigator.userAgent.indexOf('Firefox') !== -1;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * There are different ways to inject scriptlets ("push" vs "pull").
+ * This function should decide based on the environment what to use:
+ *
+ * 1) "Pushing" means the adblocker will listen on "onCommitted" events
+ *    and then execute scripts by running the tabs.executeScript API.
+ * 2) "Pulling" means the adblocker will inject a content script, which
+ *    runs before the page loads (and on the DOM changes), fetches
+ *    scriplets from the background and runs them.
+ *
+ * Note:
+ * - the "push" model requires permission to the webNavigation API.
+ *   If that is not available, the implementation will fall back to the
+ *   "pull" model, which does not have this requirement.
+ */
+function usePushScriptsInjection() {
+  // There is no fundamental reason why it should not work on Firefox,
+  // but given that there are no known issues with Firefox, let's keep
+  // the old, proven technique until there is evidence that changes
+  // are needed.
+  //
+  // Take YouTube as an example: on Chrome (or forks like Edge), the adblocker
+  // will sometimes fail to block ads if you reload the page multiple times;
+  // on Firefox, the same steps do not seem to trigger any ads.
+  return !isFirefox();
```

Comment:
there is no need to evaluate it multiple times

Reply:
It is only evaluated for the construction of the context, which is a rare operation. We can introduce a constant, of course, but I don't think it is performance sensitive.